Documentation/app-operations.md

```diff
@@ -0,0 +1,234 @@
+so that it is ready to be started by the stage1 (e.g. systemd for the default flavor).
+
+### Stage1 Default Flavor Example Workflow
+
+1. Render the application rootfs.
+2. Prepare the application/s
+3. Start the injected application using the `app start` entrypoint.
+
+This approach is similar to using `rkt run`, instead of creating a new pod,
+an existing pod will be used.
+
+## rkt app rm
+
+`coreos.com/rkt/stage1/app/rm`
+
+1. receive a reference to an application that resides inside the pod (running or stopped)
+2. stop the application if it's running.
+3. remove the contents of the application (rootfs) from the pod (keep the logs?) and delete references to it (e.g. service files).
+   - Delegate these steps by creating `app/install` and `app/rm` entrypoints?
+
+The responsibility of this entrypoint is to receive an app inside a pod and to remove it from the pod.
+so that it is ready to be started by the stage1 (e.g. systemd for the default flavor).
+After this operation, starting the application again is not possible and requires injecting the same
+application using the `app/add` entrypoint.
+
+### Stage1 Default Flavor Example Workflow
+
+1. Stop the application using the `app stop` entrypoint.
+2. Delete the application rootfs in `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
+3. Remove the application service files.
+4. Perform garbage collection during the execution of a pod.
+5. Reload the systemd daemon for changes to take place without disturbing the other tennant applications.
+
+## rkt app start
+
+`coreos.com/rkt/stage1/app/start`
+
+1. given a reference to an application that's in a startable state (see below), instruct the to start the application.
+
+The responsibility of this entrypoint is to start an application that is in a `Ready` or `Stopped` state, 
+which might be an app that was recently injected or an app that was stopped, instruct the stage1 to start the application.
+
+## rkt app stop
+
+`coreos.com/rkt/stage1/app/stop`
+
+1. given a reference to an application that's in a stoppable state (see below), instruct the stage1 to stop the application.
+
+The responsibility of this entrypoint is to stop an application that is `Running` by instructing the stage1. 
+
+**Note:** there is a graceful shutdown which sends the termination signal to application and waits for a grace period
+for application to terminate and then the application is shutdown if it doesn't terminate by the end of the grace period.
+
+# App States
+
+Expected set of app states are listed below:
+
+```go
+type AppState string
+
+const (
+    UnknownAppState AppState = "unknown"
+
+    PreparingAppState AppState = "preparing"
+
+	// Apps that are ready to be used by `app start`.
+	PreparedAppState AppState = "prepared"
+
+	RunningAppState AppState = "running"
+
+    // Apps stopped by `app stop`.
+    StoppingAppState AppState = "stopping"
+	StoppedAppState AppState = "stopped"
```

Comment:
The distinction between `stopped` and `exited` seems odd to me.

It's not possible to distinguish that reliably anyways (maybe it just decided to exit roughly when there was an attempt to stop it.. signal handling is unreliable, etc), and I'm not sure what value it provides to attempt to make this distinction.

Also, some odd spacing going on